### PR TITLE
Release step in the CI

### PR DESCRIPTION
**Pull request by piotrmsc**
2 additional steps added for release based on oathkeeper config

### CODE REVIEW

**Comment by piotrmsc** [comment]
Currently integration tests cannot be enabled due to the problem with running kind inside docker. 
We will think of the workaround, current possibilities : run kind directly on a machine or  get rid of kind and use binaries of apiserver.

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ test: generate fmt vet manifests
 # Start KIND pseudo-cluster
 kind-start:
 	GO111MODULE=on go get "sigs.k8s.io/kind@v0.4.0" && kind create cluster
-KUBECONFIG=$(shell kind get kubeconfig-path --name="kind")
+    KUBECONFIG=$(shell kind get kubeconfig-path --name="kind")
```
Comment (aeneasr) [inline Makefile]: This should be two tabs

File: .circleci/config.yml [hunk 2]
```diff
@@ -48,9 +48,47 @@ jobs:
...
+      - checkout
+      - run:
+          command: |
+            echo 'export DOCKER_FULL_TAG=$(echo $CIRCLE_TAG | tr '+' '_')' >> $BASH_ENV
+            echo 'export DOCKER_SHORT_TAG=$(echo $CIRCLE_TAG | cut -d '+' -f1)' >> $BASH_ENV
+            ./.circleci/release_name.bash
+            source $BASH_ENV
+      - setup_remote_docker
+      - run: docker login --username "$DOCKER_USERNAME" --password "$DOCKER_PASSWORD"
+      - run: curl -sL https://git.io/goreleaser | bash
```
Comment (aeneasr) [inline .circleci/config.yml]: Add line break for readability

```suggestion
      - run: curl -sL https://git.io/goreleaser | bash

`````

Comment (aeneasr) [inline .circleci/config.yml]: Goreleaser must be configured: https://goreleaser.com/

Alternatively just `docker build` and `docker push`

ps: The current docker file does not work with goreleaser, see: https://goreleaser.com/customization/#Docker

Comment (piotrmsc) [reply .circleci/config.yml]: I have pushed goreleaser yml config file, based on oathkeeper, however, I think that few things needs to be done on your side such as configuring GITHUB_TOKEN etc. Regarding dockerfile I tool a look at docu and compared it with oathkeeper dockcerfile and didn't notice any  missings. Could you point me ?:)

Comment (aeneasr) [reply .circleci/config.yml]: Go releaser builds docker images in such a way where it first builds the binaries for all platform, then builds a docker image which simply copies the linux binary into the container. So there is no actual build step involved.

File: Dockerfile [hunk 3]
```diff
@@ -1,26 +1,14 @@
...
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
+RUN apk add -U --no-cache ca-certificates
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:latest
-WORKDIR /
-COPY --from=builder /workspace/manager .
-ENTRYPOINT ["/manager"]
+FROM scratch
```
Comment (aeneasr) [inline Dockerfile]: Check if we need the distroless image from gcr or if scratch is enough!

Comment (piotrmsc) [reply Dockerfile]: scratch is enough :)

**Comment by aeneasr** [comment]
I have added the env vars!

**Pull request merged**